src/pretix/base/services/orders.py

```diff
@@ -1015,6 +1015,14 @@ def _recalculate_total_and_payment_fee(self):
         self.order.total += sum([f.value for f in self.order.fees.all()])
         self.order.save()
 
+    def _payment_fee_diff(self):
+        prov = self._get_payment_provider()
+        if prov:
+            old_fee = prov.calculate_fee(self.order.total)
+            new_total = self.order.total + self._totaldiff
+            if new_total != 0:
```

Comment:
This if statement is also found here:
https://github.com/pretix/pretix/blob/fc7bafe3d98b499f9ae2b2fe4f17e79552b62e4c/src/pretix/base/services/orders.py#L1002
I think this should have the same behaviour, although I don't see a reason for why this check exists.